### PR TITLE
Review and test speech-enhancement

### LINKED ISSUE #104: Task: speech-enhancement peer review and testing
### Description

Review and test speech-enhancement

### Tasks

- [x] Review
- [x] Test

### Freeform Notes

_No response_

### PR DESCRIPTION
**Pull request by ibevers**
This PR is for reviewing, updating, and adding tests to the speech enhancement task.

### CODE REVIEW

**Comment by fabiocat93** [comment]
thank you @ibevers , can you please add a brief description of what is this pr about?

**Comment by ibevers** [comment]
@fabiocat93 done

File: src/senselab/audio/tasks/speech_enhancement/api.py [hunk 1]
```diff
@@ -10,27 +11,31 @@
...
 
 def enhance_audios(
-    audios: List[Audio], 
-    model: SenselabModel = SpeechBrainModel(
-        path_or_uri="speechbrain/sepformer-wham16k-enhancement", 
-        revision="main"), 
-    device: Optional[DeviceType] = None
+    audios: List[Audio],
+    model: SenselabModel = SpeechBrainModel(path_or_uri="speechbrain/sepformer-wham16k-enhancement", revision="main"),
+    enhancer: Optional[SpeechBrainEnhancer] = None,
```
Comment (fabiocat93) [inline src/senselab/audio/tasks/speech_enhancement/api.py]: what is this? @ibevers

Comment (ibevers) [reply src/senselab/audio/tasks/speech_enhancement/api.py]: That's what I was referring to when I said:
"I've been thinking about this a bit, and I'm wondering if we could create a shared enhancer object that uses the same model using multiprocessing.Manager and update the API to accept it as a parameter. I went ahead and updated the API (easy to change it back, and it doesn't affect existing usage)."

Comment (ibevers) [reply src/senselab/audio/tasks/speech_enhancement/api.py]: Should I change it back?

Comment (ibevers) [reply src/senselab/audio/tasks/speech_enhancement/api.py]: I changed it back

File: src/senselab/audio/tasks/speech_enhancement/speechbrain.py [hunk 2]
```diff
@@ -75,10 +75,24 @@ def enhance_audios_with_speechbrain(
...
                     + str(expected_sample_rate)
                 )
-
-            # Enhance waveforms in a batch
-            enhanced_waveform = enhancer.separate_batch(audio.waveform)
-
-            audio.waveform = enhanced_waveform.reshape(1, -1)
-
-        return audios
+        enhanced_audios = []
```
Comment (fabiocat93) [inline src/senselab/audio/tasks/speech_enhancement/speechbrain.py]: hi @ibevers the problem with this code is that it doesn't work with files of different shape

**Comment by fabiocat93** [comment]
thank you @ibevers . starting from your edits and reviews, I have made more changes, added a brief doc.md file, and a getting started tutorial for speech enhancement through senselab. I appreciate the changes for running speech enhancement in batch, but the code you provided doesn't work with multiple audio samples of different shapes. we may introduce a way to make them all of the same size (probably padding is the best way). Maybe we can discuss this as a group to make the decision consistent with other tasks. For now, I reverted the code to the last functioning version, so that all tests pass and we are ready for a release this week

**Review by fabiocat93 — APPROVED** [review]

**Pull request merged**